tests/helpers/util.py

```diff
@@ -156,11 +157,15 @@ def get_unique_localhost():
 @contextmanager
 def run_subprocess(command: List[str], env: Dict[any, any] = None, **kwargs):
     # subprocess on Windows has a bug where it doesn't like Path.
-    proc = subprocess.Popen(
-        [str(c) for c in command], env=env, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, **kwargs
-    )
+    try:
+        proc = subprocess.Popen(
+            [str(c) for c in command], env=env, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, **kwargs
+        )
+    except OSError as e:
+        sys.stderr.write(f"{e}\n")
 
     get_output = pull_from_reader_in_background(proc.stdout)
+    assert proc.poll() is None, f"agent is not running:\n{get_output}"
```

Comment:
`get_output` is a function, not a string value.

Also this is super racy.  If the agent dies immediately after this check it will still go on happily.

Reply:
Also `proc` will be None if the OSError happens since it will throw the exception before `proc` gets assigned.